autolab/Autolab
# Catch invalid assessment config on import asmt tar

## Description
- Validate syntax of assessment config file upon tar import
- Add error handling logic for any `SyntaxError` caught

## Motivation and Context
Currently, if we import the tarfile of an assessment whose config file has a syntax error, the error is not caught. Furthermore, the assessment is created but throws an error every time we attempt to access it.

This PR adds checking to catch syntax errors and also avoids creating assessments if an error is found.

Closes #1529

Notes
- This only catches `SyntaxError`, as it's not really possible to catch errors such as `NameError` and `ArgumentError`
- There is still no checking for import from file system, but this can be the scope of a future PR
- I considered a few ways of displaying the error message
  - We can't just set `@error = e` and use `@error` inside `install_assessment.html.erb`. If we redirect to `install_assessment`, the instance variable disappears. If we render it directly, we miss out on logic set by the controller
  - We could create a separate page just for displaying the error, but then there will be significant duplication with `reload.html.erb`, and it detracts from the workflow
  - We can't use `reload.html.erb` directly either, since it has assessment specific logic (a button to `reload`). It wouldn't really make sense to create a partial just for this either
- Ultimately, I decided to display the error directly inside the flash message. However, this means that I can't show extra info such as the backtrace, as that would cause a cookie overflow error due to its length

## How Has This Been Tested?
- Modify an existing assessment's config file (`<asmt>.rb`) and make it invalid, for example by changing the module's name to be all lowercase
- Export the assessment tar

Before PR
- Create another course and import the assessment tar
- Observe that the assessment is created but we get an error every time we try to access it
- Also observe that the assessment folder is created (`courses/<course>/<asmt>`) 

After PR
- Delete the assessment folder
- Import the assessment tar again
- Observe that the error is caught and displayed to the user, and that the assessment is **not** created
- Also observe that the assessment folder is **not** created

<img width="1264" alt="Screenshot 2022-12-21 at 23 31 14" src="https://user-images.githubusercontent.com/9074856/209071721-ea1da88b-6b1a-4a54-8a57-4aa9dc561d20.png">

- Finally, import a valid assessment tar and ensure that it still works

## Types of changes

## Checklist:
